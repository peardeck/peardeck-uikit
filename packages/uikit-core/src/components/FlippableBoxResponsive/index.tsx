export {
  FlippableBoxResponsive,
  FlippableBoxResponsiveProps,
} from "./FlippableBoxResponsive";
