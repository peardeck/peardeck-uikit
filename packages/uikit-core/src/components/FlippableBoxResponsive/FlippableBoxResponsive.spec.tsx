import { create } from "react-test-renderer";
import React from "react";
import { FlippableBoxResponsive } from "./FlippableBoxResponsive";

describe("FlippableBoxResponsive", () => {
  test("it matches the snapshot", () => {
    const json = create(
      <FlippableBoxResponsive
        front={<div>This is the front</div>}
        back={<div>This is the back</div>}
      />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  describe("flipped", () => {
    test("it matches the snapshot", () => {
      const json = create(
        <FlippableBoxResponsive
          front={<div>This is the front</div>}
          back={<div>This is the back</div>}
          flipped
        />
      ).toJSON();
      expect(json).toMatchSnapshot();
    });
  });
});
