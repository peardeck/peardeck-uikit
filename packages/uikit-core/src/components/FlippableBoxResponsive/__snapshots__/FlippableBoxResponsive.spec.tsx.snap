// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlippableBoxResponsive flipped it matches the snapshot 1`] = `
.c0 {
  position: relative;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
}

.c0 .sc-AxirZ {
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c1 {
  position: relative;
  z-index: 2;
  -webkit-transform: rotateY(0deg);
  -ms-transform: rotateY(0deg);
  transform: rotateY(0deg);
  -webkit-animation-name: kvSYFG;
  animation-name: kvSYFG;
  height: 100%;
  width: 100%;
  -webkit-animation-name: none;
  animation-name: none;
}

.c2 {
  position: relative;
  z-index: 1;
  -webkit-transform: rotateY(-180deg);
  -ms-transform: rotateY(-180deg);
  transform: rotateY(-180deg);
  -webkit-animation-name: bjQFEz;
  animation-name: bjQFEz;
  height: 0;
  width: 0;
  visibility: collapse;
  -webkit-animation-name: none;
  animation-name: none;
}

<div
  className="sc-AxjAm c0"
>
  <div
    className="sc-AxjAm sc-AxirZ c1"
  >
    <div>
      This is the front
    </div>
  </div>
  <div
    className="sc-AxjAm sc-AxirZ c2"
  >
    <div>
      This is the back
    </div>
  </div>
</div>
`;

exports[`FlippableBoxResponsive it matches the snapshot 1`] = `
.c0 {
  position: relative;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
}

.c0 .sc-AxirZ {
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
}

.c1 {
  position: relative;
  z-index: 2;
  -webkit-transform: rotateY(0deg);
  -ms-transform: rotateY(0deg);
  transform: rotateY(0deg);
  -webkit-animation-name: iWmnfy;
  animation-name: iWmnfy;
  height: 100%;
  width: 100%;
  -webkit-animation-name: none;
  animation-name: none;
}

.c2 {
  position: relative;
  z-index: 1;
  -webkit-transform: rotateY(-180deg);
  -ms-transform: rotateY(-180deg);
  transform: rotateY(-180deg);
  -webkit-animation-name: cToBvn;
  animation-name: cToBvn;
  height: 0;
  width: 0;
  visibility: collapse;
  -webkit-animation-name: none;
  animation-name: none;
}

<div
  className="sc-AxjAm c0"
>
  <div
    className="sc-AxjAm sc-AxirZ c1"
  >
    <div>
      This is the front
    </div>
  </div>
  <div
    className="sc-AxjAm sc-AxirZ c2"
  >
    <div>
      This is the back
    </div>
  </div>
</div>
`;
